datahub/dnb_api/tasks.py

```
@@ -21,6 +22,8 @@ def _sync_company_with_dnb(company_id, fields_to_update, task):
         if is_server_error(exc.status_code):
             raise task.retry(exc=exc, countdown=60)
         raise
+    except DNBServiceConnectionError as exc:
```

I think we should also consider that we get 404s from Cloud Foundry when there are routing problems (route unmapped, app still starting etc.) but these are only transient as well.

---

Just to talk this through...  In this case I believe that the block will trigger here: https://github.com/uktrade/data-hub-api/pull/2314/files/fbaaefeff61e9ea37a13dc4b6f9373cbff40097b#diff-f3cf320eaf9fd1b399380f387051a25aR93

The exception class/error is probably a little misleading (it's not strictly a service error) but I feel like it's OK to flag this as a logged exception even when the vast majority of 404s might be from a transient routing problem.  This will also flag up the possible situation where the endpoint URL has changed on dnb-service, etc.

My feeling is that the application is not the place to solve (or suppress errors for) an ops-specific oddity.  Are there other considerations that I'm missing here?